common/components/Header/components/NetworkSelector/NetworkOption.tsx

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import { translateRaw } from 'translations';
+import classnames from 'classnames';
+import { isAutoNode, isAutoNodeConfig } from 'libs/nodes';
+import { NodeConfig } from 'types/node';
+import { NetworkConfig } from 'types/network';
+import NodeOption from './NodeOption';
+import './NetworkOption.scss';
+
+interface Props {
+  nodes: NodeConfig[];
+  network: NetworkConfig;
+  nodeSelection: string;
+  isNetworkSelected: boolean;
+  isExpanded: boolean;
+  selectNode(node: NodeConfig): void;
+  selectNetwork(network: NetworkConfig): void;
+  toggleExpand(network: NetworkConfig): void;
+}
+
+export default class NetworkOption extends React.PureComponent<Props> {
+  public render() {
+    const { nodes, network, nodeSelection, isExpanded, isNetworkSelected } = this.props;
+    const borderLeftColor = network.isCustom ? '#CCC' : network.color;
+    const singleNodes = nodes.filter(node => !isAutoNodeConfig(node));
+    const isAutoSelected = isNetworkSelected && isAutoNode(nodeSelection);
+
+    return (
+      <div className="NetworkOption" style={{ borderLeftColor }}>
+        <div className="NetworkOption-label">
+          <div
+            className={classnames({
+              'NetworkOption-label-name': true,
+              'is-selected': isNetworkSelected,
+              'is-specific-node': isNetworkSelected && !isAutoSelected && singleNodes.length > 1
+            })}
+            title={translateRaw('NETWORKS_SWITCH', { $network: network.name })}
+            onClick={this.handleSelect}
+          >
+            {network.name}
+          </div>
+          <button
+            className={classnames('NetworkOption-label-expand', isExpanded && 'is-expanded')}
+            onClick={this.handleToggleExpand}
+          >
+            <i className="fa fa-chevron-down" />
+          </button>
+        </div>
+        {isExpanded && (
+          <div className="NetworkOption-nodes">
+            {singleNodes.map(node => (
+              <NodeOption
+                key={node.id}
+                node={node}
+                isSelected={node.id === nodeSelection}
+                isAutoSelected={isAutoSelected}
+                select={this.props.selectNode}
+              />
+            ))}
+          </div>
+        )}
+      </div>
+    );
+  }
+
+  private handleSelect = () => {
+    this.props.selectNetwork(this.props.network);
```

Comment:
You could instead make selectNetwork a HOF in the parent component to avoid making extra methods here (and actually you could convert this entire component to a HOF by doing so)